### PR TITLE
[INLONG-6387][Manager] Adapt auto-push source status and fix source field update

### LINKED ISSUE #6387: [Improve][Manager] Adapt auto-push source type status and fix source field update
### Description

Problems:
- The auto push source type can be in status that does not make sense, for example, TO_BE_ISSUED_FROZEN.
- The auto push source fields are not correctly updated.
- Other adaptation issue.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fixes #6387 

### Motivation
- The auto push source type can be in status that does not make sense, for example, TO_BE_ISSUED_FROZEN.
- The auto push source fields are not correctly updated.
- Other adaptation issue.

### Modification
- Replace all temporary status with end status for auto push source type.
- Correctly update stream source fields and inlong stream fields.
- Remove stream operation on sink update (**open for discussion**)
     _So what's the appropriate time to restart after config updates ? Perhaps it's better be launched by end user in a separate call. According to the Single Responsibility Principle the service update handler should only perform no other than config updating._

### CODE REVIEW

**Review by ciscozhou — APPROVED** [review]
LGTM

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java [hunk 1]
```diff
@@ -254,15 +238,6 @@ public Boolean update(SinkRequest request, String operator) {
         StreamSinkOperator sinkOperator = operatorFactory.getInstance(request.getSinkType());
         sinkOperator.updateOpt(request, nextStatus, operator);
 
-        // If the stream is [CONFIG_SUCCESSFUL], then asynchronously start the [CREATE_STREAM_RESOURCE] process
```
Comment (woofyzhao) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java:257]: So what's the appropriate time to restart after config updates ? Perhaps it's better be launched by end user in a separate call. According to the Single Responsibility Principle the service update handler should only perform no other than config updating.

**Comment by woofyzhao** [comment]
Needs further discussion. Close first.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java [hunk 2]
```diff
@@ -125,15 +118,6 @@ public Integer save(SinkRequest request, String operator) {
             sinkEntity.setStatus(nextStatus.getCode());
             sinkMapper.updateStatus(sinkEntity);
         }
-        // If the stream is [CONFIG_SUCCESSFUL], then asynchronously start the [CREATE_STREAM_RESOURCE] process
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java:128]: Suggest adding a param to control whether to enable the process.

I will commit it in a new PR.

**Review by healchow — APPROVED** [review]
+1

**Pull request merged**